src/inmemory.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * In-memory version of the `__aux` wrapper used in entity.js
+ */
+
+var _      = require('lodash');
+var crypto = require('crypto');
+
+// the in-memory data; stored globally as this is a better model for
+// Azure than storing each table as an instance or class property
+var tables = {};
+
+var InMemoryWrapper = function InMemoryWrapper(table) {
+  this.table = table;
+};
+
+/* Internal utilities */
+
+var entityEtag = function(entity) {
+  var sha1 = crypto.createHash('sha1');
+  // filter out metadata
+  var odata = /^odata\.|@odata/;
+  entity = _.omit(entity, function(v, k) { return odata.test(k); });
+  sha1.update(JSON.stringify(entity));
+  return sha1.digest('hex');
+};
+
+var makeError = function(statusCode, code) {
+  var err = new Error(code);
+  err.statusCode = statusCode;
+  err.code = code;
+  return err;
+};
+
+var makeKey = function(partitionKey, rowKey) {
+  return partitionKey.replace(/!/g, '!!') + '!' + rowKey.replace(/!/g, '!!');
+};
+
+/**
+ * Create table.
+ *
+ * @method createTable
+ * @return {Promise} A promise that the table was created.
+ */
+InMemoryWrapper.prototype.createTable = function() {
+  tables[this.table] = {};
+  // TODO: EntityAlready Exists?
+  return Promise.resolve();
+};
+
+/**
+ * Delete table.
+ *
+ * @method deleteTable
+ * @return {Promise} A promise that the table was marked for deletion.
+ */
+InMemoryWrapper.prototype.deleteTable = function() {
+  // TODO: 404?
```

Comment:
Again, yes... we do handle the 404 error... IMO it's stupid of azure to have it, because delete should be idempotent... but apparently they like footguns. So doing 404 would increase our test coverage.